localstack-core/localstack/utils/aws/arns.py

```diff
@@ -196,12 +196,13 @@ def dynamodb_stream_arn(
 
 def cloudwatch_alarm_arn(alarm_name: str, account_id: str, region_name: str) -> str:
     pattern = "arn:%s:cloudwatch:%s:%s:alarm:%s"
-    return _resource_arn(alarm_name, pattern, account_id=account_id, region_name=region_name)
+    # format pattern directly as alarm_name can include ":" and this is not supported by the helper _resource_arn
+    return pattern % (get_partition(region_name), region_name, account_id, alarm_name)
```

Comment:
Good catch and nice fix!

nit: you could also use f-strings here
```suggestion
    # format pattern directly as alarm_name can include ":" and this is not supported by the helper _resource_arn
    return f"arn:{get_partition(region_name)}:cloudwatch:{region_name}:{account_id}:alarm:{alarm_name}"
```

Reply:
sure, will do :)